### PR TITLE
Fix typos and minor style problems

### PR DESCRIPTION
**Pull request by tkruse**

### CODE REVIEW

**Comment by blakehawkins** [comment]
All looks good.

File: CppCoreGuidelines.md [hunk 1]
```diff
@@ -12376,18 +12378,20 @@ Aternatively, we will decide that no change is needed and delete the entry.
 * What to do with leaks out of temporaries? : `p = (s1+s2).c_str();`
 * pointer/iterator invalidation leading to dangling pointers
 
-	void bad()
-	{
-		int* p = new int[700];
-		    int* q = &p[7];
-    	delete p;
+```
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: For now, we are trying to avoid fencing.  Could you rewrite this part of the patch with indentation only?
Thanks!

**Comment by tkruse** [comment]
rebased, removed fence

**Comment by tkruse** [comment]
rebased

**Comment by blakehawkins** [comment]
Looks good.

File: CppCoreGuidelines.md [hunk 2]
```diff
@@ -3378,7 +3378,7 @@ The standard library requires that all classes it deals with have destructors th
 None have succeeded to come up with a general scheme.
 This can be be a real practical problem: For example, what about a sockets that won't close?
 The writer of a destructor does not know why the destructor is called and cannot "refuse to act" by throwing an exception.
-See <a =href="#Sd-dtor">discussion</a>.
+See [discussion](#Sd dtor).
```
Comment (gdr-at-ms) [inline CppCoreGuidelines.md]: This undoes a fix I applied earlier.  Can we keep it consistently with the constructor case?

Comment (tkruse) [inline CppCoreGuidelines.md]: Duh, it undoes a fix I provided myself. I assume you talk about the dash? Fixed that one. and pushed again.

**Comment by tkruse** [comment]
rebased

**Pull request merged**